dxtbx/imageset.py

```diff
@@ -148,6 +148,61 @@ def paths(self):
     return [self.get_path(i) for i in range(len(self))]
 
 
+class ImageSetLazy(ImageSet):
```

Comment:
A while back, I did some work to try and put the ImageSet/ImageSweep classes into C++ so that the image data could be read both in C++ and Python. It looks like ImageSetLazy just has a python interface at the moment. As it inherits from the ImageSet class, passing an ImageSetLazy object into a C++ function expecting an ImageSet object should work in that it will treat it as an ImageSet object. 

However, I'm just wondering what will happen if the C++ function calls get_detector on this object? I don't think these overloaded methods won't be called unless we do something in boost python (https://www.boost.org/doc/libs/1_61_0/libs/python/doc/html/tutorial/tutorial/exposing.html#tutorial.exposing.class_virtual_functions). I suppose one question would be whether we expect this to be needed; however, I think it would be desirable to ensure that the ImageSetLazy class also works with C++ functions.

Reply:
You are correct about get_detector not working with ImageSetLazy through C++ unless the python version has been called first.  However, we grepped the sources for this use case and didn't find it.